Sources/Sentry/SentryScope.m

```diff
@@ -59,26 +59,33 @@ @interface SentryScope ()
  */
 @property(atomic) enum SentryLevel levelEnum;
 
+@property(atomic) NSInteger maxBreadcrumbs;
+
 @end
 
 @implementation SentryScope
 
 #pragma mark Initializer
 
-- (instancetype)init {
+- (instancetype)initWithMaxBreadcrumbs:(NSInteger)maxBreadcrumbs {
     if (self = [super init]) {
         self.listeners = [NSMutableArray new];
+        self.maxBreadcrumbs = maxBreadcrumbs;
         [self clear];
         // Default values
         NSDictionary *infoDict = [[NSBundle mainBundle] infoDictionary];
         if (nil != infoDict) {
             [self setRelease:[NSString stringWithFormat:@"%@@%@+%@", infoDict[@"CFBundleIdentifier"], infoDict[@"CFBundleShortVersionString"],
-                infoDict[@"CFBundleVersion"]]];
+                                                        infoDict[@"CFBundleVersion"]]];
         }
     }
     return self;
 }
 
+- (instancetype)init {
+    return [self initWithMaxBreadcrumbs:100];
```

Comment:
Default could be set on some const later, but it's 100 ^

Reply:
This has to somehow be picked up from the Sentry Options OR we consider this the absolute maximum because we use the option elsewhere.